demo/VPlay.hs

```diff
@@ -195,18 +197,21 @@ videoPlayer cfg src = do
 
   (renderTexture, getTexture, cleanup) <- textureReader src
 
+  -- First frame begins.
+  timeRef <- liftIO . newIORef =<< SDL.time
+
   liftIO $ whileJust_ (nothingOnQuit getTexture) $
     \ (next, time) -> do
 
       {- Rendering. -}
 
       -- Rendering start time.
-      rStartTime <- SDL.time
+      rStartTime <- liftIO $ readIORef timeRef
 
       renderTexture next
 
       -- Finish time of rendering.
-      rFinishTime <- SDL.time
+      rFinishTime <- SDL.time :: IO Double
```

Comment:
Without the annotation, the types of `timeRef`, `rStartTime` and `rFinishTime` are ambiguous.